modules/core/src/main/java/org/apache/ignite/internal/processors/cache/GridCacheMapEntry.java

```diff
@@ -1723,10 +1723,12 @@ protected Object keyValue(boolean cpy) {
             if (metrics && cctx.statisticsEnabled()) {
                 cctx.cache().metrics0().onRemove();
 
-                T2<GridCacheOperation, CacheObject> entryProcRes = tx.entry(txKey()).entryProcessorCalculatedValue();
+                if (tx != null) {
```

Comment:
Does this change correspond to the original issue?

Reply:
No. I've reverted it and created https://ggsystems.atlassian.net/browse/GG-37743